proxy/http/HttpTunnel.cc

```diff
@@ -1332,6 +1332,9 @@ HttpTunnel::consumer_handler(int event, HttpTunnelConsumer *c)
     if (c->producer && c->producer->handler_state == 0) {
       if (event == VC_EVENT_WRITE_COMPLETE) {
         c->producer->handler_state = HTTP_SM_POST_SUCCESS;
+        // If the consumer completed, presumably the producer successfully read and is done
+        c->producer->read_success = true;
```

Comment:
Note: This seems the key change, forcing to make the producer done on the `VC_EVENT_WRITE_COMPLETE` event.

Reply:
So you found and fixed an issue i came across...

You merged it, and flagged my work? May i ask why per say..? I've only ever contributed to branches. Never flagged and had deleted..?

Reply:
Sorry, not sure what you mean by flagging your work.